stan/math/prim/mat/fun/log_sum_exp.hpp

```diff
@@ -25,6 +26,8 @@ namespace math {
 template <int R, int C>
 double log_sum_exp(const Eigen::Matrix<double, R, C>& x) {
   const double max = x.maxCoeff();
+  if (!boost::math::isfinite(max))
```

Comment:
I think the preference is to use standard library functions rather than boost where possible